controllers/addresspool_controller.go

```diff
@@ -146,6 +148,22 @@ func (r *AddressPoolReconciler) Reconcile(ctx context.Context, request ctrl.Requ
 		return reconcile.Result{}, err
 	}
 
+	// Restore the data network status
+	if r.checkRestoreInProgress(instance) {
+		r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "Restoring '%s' Addresspool resource status without doing actual reconciliation", instance.Name)
+		if err := r.RestoreDataNetworkStatus(instance); err != nil {
```

Comment:
RestoreAddressPool Status function have to be defined and call that function here

Reply:
Have updated the func called